### PR TITLE
Add docs for images, command exec, port forwarding

### PR DESCRIPTION
**Pull request by ncdc**
Add docs for images, image registries, HTTP command execution, and port
forwarding.

### CODE REVIEW

File: architecture/http_remote_command_execution.adoc [hunk 1]
```diff
@@ -9,4 +9,53 @@
 
 toc::[]
 
-== Overview
\ No newline at end of file
+== Overview
+
+OpenShift takes advantage of a feature built into Kubernetes to support executing commands in containers. This is implemented using HTTP along with a multiplexed streaming protocol such as link:http://www.chromium.org/spdy[SPDY] or link:https://http2.github.io/[HTTP/2].
```
Comment (nhr) [inline architecture/http_remote_command_execution.adoc]: No action required, but fyi you do not need to use `link:` when your URL starts with a well-known proto. So you could just use `http://foo.bar[link text]` in these instances.

Comment (ncdc) [inline architecture/http_remote_command_execution.adoc]: Thanks, I'm an asciidoc n00b

Comment (ncdc) [inline architecture/http_remote_command_execution.adoc]: Also, there are other docs using Markdown syntax `[text](link)` that should probably be converted

File: architecture/image_registry.adoc [hunk 2]
```diff
@@ -10,4 +10,21 @@
...
+
+=== 3rd Party Registries
+
+OpenShift can create containers using images from 3rd party registries, but these registries likely won't offer the same image notification support that the integrated OpenShift registry does. If not, OpenShift is able to poll the other registries for changes to image repositories. When new images are detected, the same build/deployment/etc reactions described above occur. (NOTE: polling is not implemented yet.)
+
+==== Authentication
+
+OpenShift is able to communicate with registries to access private image repositories using credentials supplied by the user. This allows OpenShift to push and pull images to/from private repositories.
+
+NOTE: This isn't implemented yet.
```
Comment (nhr) [inline architecture/image_registry.adoc]: This line will actually create a NOTE block, in contrast with your use of NOTE in lines 18 and 24 where it is just going to be rendered as text. Please be consistent; in this case I'd suggest taking this sentence and throwing it in parenthesis at the end of the paragraph in line 28 (the same way you do in line 24). That will prevent the fancy rendering.

File: architecture/openshift_model.adoc [hunk 3]
```diff
@@ -45,10 +45,16 @@ The custom build strategy is very similar to *Docker build* strategy, but users
 == DeploymentConfig
 
 == Image
+OpenShift stores information about Docker images including the "pull spec" (what you'd use to pull the image) and complete metadata about the image (e.g. command, entrypoint, environment variables, etc.). Images in OpenShift are immutable.
```
Comment (nhr) [inline architecture/openshift_model.adoc]: These explanations of OpenShift objects is fine as-is, but if it is easy to link to / include the JSON structure of each, I think that would be valuable. I tried to do the same for the k8s objects.

File: architecture/image_registry.adoc (hunk 2 above)
Comment (ncdc) [inline architecture/image_registry.adoc]: Fixed

File: architecture/openshift_model.adoc (hunk 3 above)
Comment (ncdc) [inline architecture/openshift_model.adoc]: I'll work on getting those, sure.

Comment (ncdc) [inline architecture/openshift_model.adoc]: JSON examples added.

**Comment by nhr** [comment]
Looks good to me.

**Pull request merged**